File: solutions/ruby/raindrops/1/raindrops.rb
module Raindrops 
  def self.convert(number) 
    result = ""
    return number.to_s if (number % 3 != 0 && number % 5 != 0 && number % 7 != 0 )
    result += "Pling" if (number % 3 == 0)
    result += "Plang" if (number % 5 == 0)
    result += "Plong" if (number % 7 == 0)
    result
  end
end

File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"

namespace raindrops {
    std::string convert(int number) {
        std::string res = "";
        if (number % 3 == 0 || number % 5 == 0 || number % 7 == 0) {
            if (number % 3 == 0) {
                res += "Pling";
            } 
            
            if (number % 5 == 0) {
                res += "Plang";
            } 
            
            if (number % 7 == 0) {
                res += "Plong";
            } 
        } else {
            res += std::to_string(number);
        }
        
        return res;
    }
}  // namespace raindrops
